vivisect/__init__.py

```diff
@@ -1170,17 +1170,25 @@ def parseOpcode(self, va, arch=envi.ARCH_DEFAULT, skipcache=False):
             loctup = self.getLocation(va)
             # XXX - in the case where we've set a location on what should be an
             # opcode lets make sure L_LTYPE == LOC_OP if not lets reset L_TINFO = original arch param
-            # so that at least parse opcode wont fail
+            # so that at least parse opcode won't fail
             if loctup is not None and loctup[L_TINFO] and loctup[L_LTYPE] == LOC_OP:
                 arch = loctup[L_TINFO]
+
+        amod = self.imem_archs[(arch & envi.ARCH_MASK) >> 16]
+        # TODO: Consider reserving another key so architectures can pass down info specific to them
+        extra = {
+            'platform': self.getMeta('Platform')
+        }
```

Comment:
can this be something we have cached?  creating a dict, with a call to getMeta() for every single opcode will have an unnecessary performance impact.

Reply:
So fun fact: I actually tested this idea. In order to keep the dictionary around, I used set/getTransMeta (to avoid the event stream and keep lower overhead). 

After some kernprof inspection, turns out there's not really much of a difference. My decoding loop was fairly tight but still had ~1M iterations in the loop. But across several runs, there was only a few hundredths of a second difference between either implementation (which one was faster was more up to the linux scheduler and what else was running on my machine at the time).

But ultimately you're trading a function calls or two plus an existence check for 1 unconditional dictionary construction and 1 function call. The performance shook out to be the same and I felt this was slightly more expressive with its avoidance of branching logic.

Side note: this was on python 3.9. I know 3.10 and 3.11 are different, I just haven't run them.

Reply:
i'm not asking for anything that special.  this isn't a branching check, it's simply "can you please create the dict outside the loop?"
set it as a class field of the workspace.  simply move the code elsewhere so the creation event happens once (or fewer).
does that make sense?  or am i missing something?

Reply:
nm, i see the complexity this could entail.
we could create a `VivWorkspace._cached_parser_context` field that always exists if you're using this `parseOpcode` but we can visit that later if it makes sense.